src/main/java/org/killbill/billing/plugin/adyen/client/AdyenConfigProperties.java

```diff
@@ -225,8 +228,20 @@ public AdyenConfigProperties(final Properties properties, final String currentRe
 
         this.rbacUsername = properties.getProperty(PROPERTY_PREFIX + "rbacUsername");
         this.rbacPassword = properties.getProperty(PROPERTY_PREFIX + "rbacPassword");
-        this.apiKey = properties.getProperty(PROPERTY_PREFIX + "apiKey");
-        this.environment = properties.getProperty(PROPERTY_PREFIX + "environment");
+
+        this.environment = properties.getProperty(CHECKOUT_PREFIX + "environment", "TEST");
+        readCheckoutApiKeyConfig(properties);
+    }
+
+    private void readCheckoutApiKeyConfig(final Properties properties) {
+        readConfigurationValuesToList(properties.getProperty(CHECKOUT_PREFIX + "country"), checkoutApiCountries);
```

Comment:
This looks bad.
Declaring `checkoutApiCountries` as instance variable without initialisation and populating it as part of `readCheckoutApiKeyConfig` call by passing the `checkoutApiCountries` instance variable to instance function.

Refactor this.

Reply:
readConfigurationValuesToList is existing function that takes list variable to be populated. Passing the argument to that function makes is usable for any instance variable with similar config value. Otherwise, code will be replicated for each instance variable initialisation. Hope it clarifies. Below is example of other usages -

>         readConfigurationValuesToList(properties.getProperty(PROPERTY_PREFIX + "sensitiveProperties"), sensitivePropertyKeys);
>         readConfigurationValuesToList(properties.getProperty(PROPERTY_PREFIX + "persistablePluginProperties"), persistablePluginProperties);